internal/blockchain/ethereum/ethereum.go

```diff
@@ -836,11 +836,25 @@ func (e *Ethereum) QueryContract(ctx context.Context, signingKey string, locatio
 	if err != nil || !res.IsSuccess() {
 		return nil, err
 	}
-	output := &queryOutput{}
-	if err = json.Unmarshal(res.Body(), output); err != nil {
+
+	queryOutput := make(map[string]interface{})
+	if err = json.Unmarshal(res.Body(), &queryOutput); err != nil {
```

Comment:
If I understand correctly, the proposal here is:
- If the ABI has a single **unnamed** parameter, then we return `output: {whatever}`
- If the ABI has a single **named** parameter, then we return `output: {name1: {whatever}}`
- If the ABI has multiple **unnamed** parameters, then we return `output: {output: {whatever1}, output1: {whatever2}}`
- If the ABI has multiple **named** parameters, then we return `output: {name1: {whatever1}, name2: {whatever2}}`

Reply:
Almost there just this one is different:
-  If the ABI has multiple unnamed parameters, then we return `output: {whatever1}, output1: {whatever2}` instead of it being nested. 

Discussed this one with @Chengxuan and thought it would be better as above but happy to change

Reply:
ok, so the logic is "if any of the data comes back from the connector contains `output` as a field, return the whole output unchanged

Reply:
... otherwise, nest whatever came back from the connector under an `output` parent object

Reply:
yes

Reply:
Fabric just goes with the approach of proxying data https://hyperledger.github.io/firefly/head/tutorials/custom_contracts/fabric.html#request-3

Reply:
I'm worried about proxying the data across since we would potentially break users but actually they are already broken by this!

If we proxy:

If the ABI has a single unnamed parameter, then we return `{output: {whatever}}`
If the ABI has a single named parameter, then we return `{name1: {whatever}}`
If the ABI has multiple unnamed parameters, then we return `{output: {whatever1}, output1: {whatever2}}`
If the ABI has multiple named parameters, then we return` {name1: {whatever1}, name2: {whatever2}}`

Reply:
That Fabric doc must be out of date then! 

It's still the same problem I believe. We are decoding to an object that expects a key "Result" https://github.com/hyperledger/firefly/blob/e2f00f021e1abca467532ca6dbd4f8a804091827/internal/blockchain/fabric/fabric.go#L102 so the fabric connector needs to return that. Either the fabric connector is incapsulating everything under that result field or it would hit the same error with multiple values

Reply:
I will amend this PR to proxy the direct output for ethereum connector and we can discuss the fabric one as well and decide in a future PR - need to look at the fabric connector code as well

Reply:
Fabric construct that object after serialisation https://github.com/hyperledger/firefly-fabconnect/blob/b46951bb6613a2981b89c3ea2e69048441ec41ed/internal/rest/sync/syncdispatcher.go#L200-L203 completely different to ethereum where we add a custom name generator to override the serialisation

Reply:
> so the fabric connector needs to return that

Yes, I agree `result` is where Fabric connector returns it. But that's just an `interface{}`.

e.g. - in Ethereum **today** which is broken:

- The connector returns the result _at the top level_ of the JSON, and the result must have exactly one object we care about called `output`

- The connector returns the result _in a sub-field_ of the JSON called `result`, and can specify anything it likes there which we will return _at the top level_ of the JSON to the end user

Reply:
So, I think if you change Ethereum as follows, it will be very consistent with Fabric:

```go
	var output interface{}
	if err = json.Unmarshal(res.Body(), &output); err != nil {
		return nil, err
	}
	return output, nil // note UNLIKE fabric this is just `output`, not `output.Result` - but either way the top level of what we return to the end user, is whatever the Connector sent us
```